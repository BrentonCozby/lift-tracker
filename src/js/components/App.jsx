import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { rootUrl } from '../../../config.js'

import MenuContainer from '../containers/MenuContainer.jsx'
import NoMatch from './NoMatch.jsx'
import Footer from './Footer.jsx'
import HomeContainer from '../containers/HomeContainer.jsx'
import LoginPageContainer from '../containers/LoginPageContainer.jsx'
import ProgramDetailContainer from '../containers/ProgramDetailContainer.jsx'

const App = ({ isLoggedIn }) => (
    <div className="App">
        <MenuContainer />
        <Switch>
            <Route exact path={`${rootUrl}`} component={HomeContainer} />
            <Route exact path={`${rootUrl}login`} component={LoginPageContainer} />
            <Route path={`${rootUrl}programs/:id`} exact component={({ match }) => {
                return <ProgramDetailContainer programId={match.params.id} />
            }} />
            <Route component={NoMatch}/>
        </Switch>
        <Footer />
    </div>
)

export default App
