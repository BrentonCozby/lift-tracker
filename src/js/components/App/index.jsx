import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
    listenForAuthStateChanged,
    retrieveLoginResult
} from '../../actions/user.js'

import View from './view.jsx'

class App extends Component {

    componentWillMount() {
        this.props.retrieveLoginResult()
        this.props.listenForAuthStateChanged()
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.isLoggedIn === false && nextProps.location.pathname !== '/login') {
            this.props.history.replace(`${PP}login`)
        }
    }

    render() {
        return (
            <View
                isLoggedIn={this.props.isLoggedIn}
            />
        )
    }
}

const mapStateToProps = function(state) {
    return {
        isLoggedIn: state.user.isLoggedIn
    }
}

export default connect(mapStateToProps, {
    listenForAuthStateChanged,
    retrieveLoginResult
})(App)
