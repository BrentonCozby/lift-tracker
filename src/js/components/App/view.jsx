import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { StripeProvider } from 'react-stripe-elements'

import Menu from '../Menu/index.jsx'
import NoMatch from '../NoMatch/index.jsx'
import Footer from '../Footer/index.jsx'
import HomePage from '../HomePage/index.jsx'
import LoginPage from '../LoginPage/index.jsx'
import ProgramDetail from '../ProgramDetail/index.jsx'
import PaymentPage from '../PaymentPage/index.jsx'
import LoadingOverlay from '../LoadingOverlay/index.jsx'

export default ({
    isLoggedIn
}) => (
    <StripeProvider apiKey={process.env.perishable_key}>
        <div className="App">
            <Menu />
            <LoadingOverlay />
            <Switch>
                <Route exact path={PP} component={HomePage} />
                <Route exact path={`${PP}login`} component={LoginPage} />
                <Route exact path={`${PP}programs/:id`} component={({ match }) => {
                    return <ProgramDetail programId={match.params.id} />
                }} />
                <Route exact path={`${PP}payment`} component={PaymentPage} />
                <Route component={NoMatch}/>
            </Switch>
            <Footer />
        </div>
    </StripeProvider>
)
