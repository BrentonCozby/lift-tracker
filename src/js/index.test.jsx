import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import configStore from './store.js'
import App from './components/app-component.jsx'

describe('app-component', () => {
    let stripeMockFn
    let stripeMockResult

    beforeEach(() => {
        stripeMockResult = {
            elements: jest.fn(),
            createToken: jest.fn(),
            createSource: jest.fn(),
        }
        stripeMockFn = jest.fn().mockReturnValue(stripeMockResult)
        window.Stripe = stripeMockFn

        process.env.perishable_key = 'TEST'
    })
    
    test('renders App with router and store without crashing', () => {
        const store = configStore()
        const reactStripeElements = jest.genMockFromModule('react-stripe-elements')

        reactStripeElements.StripeProvider = jest.fn()

        const wrappedApp = renderer.create(
            <Provider store={store}>
                <Router>
                    <Route path={PP} component={App} />
                </Router>
            </Provider>
        )

        expect(wrappedApp.toJSON()).toMatchSnapshot()
    })
})