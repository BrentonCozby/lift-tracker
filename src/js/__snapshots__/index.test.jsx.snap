// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-component renders App with router and store without crashing 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="menu-overlay"
      onClick={[Function]}
    />
    <div
      className="menu-button"
      onClick={[Function]}
    >
      <span
        className="bar"
      />
      <span
        className="bar"
      />
      <span
        className="bar"
      />
    </div>
    <div
      className="Menu"
    >
      <a
        className="Menu-item"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="Menu-item"
        href="login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </div>
  <div
    className="LoadingOverlay visible"
  >
    <img
      alt="loading spinner"
      className="spinner"
      src="test-file-stub"
    />
  </div>
  <div
    className="Home"
  >
    <h2
      className="app-title"
    >
      Lift Tracker
    </h2>
    <table
      className="ProgramsList"
    >
      <thead>
        <tr>
          <th>
            <h2
              className="ProgramsList-title"
            >
              Programs
            </h2>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
  <footer
    className="Footer"
  >
    <p
      className="copyright"
    >
      Â© 2017. Brenton Cozby. View more projects at 
      <a
        href="https://brentoncozby.com"
      >
        brentoncozby.com
      </a>
      .
    </p>
  </footer>
</div>
`;
